code/controllers/subsystem/dbcore.dm

```diff
@@ -170,17 +170,26 @@ SUBSYSTEM_DEF(dbcore)
 	connection = SSdbcore.connection
 
 /datum/controller/subsystem/dbcore/Shutdown()
+	to_chat(world, span_boldannounce("Clearing DB queries standby:[queries_standby_num] active: [queries_active_num]"))
 	//This is as close as we can get to the true round end before Disconnect() without changing where it's called, defeating the reason this is a subsystem
 	if(SSdbcore.Connect())
+		//Execute all waiting queries
 		for(var/datum/db_query/query in queries_standby)
-			run_query(query)
+			//In theory these would be async anyway as MC processing should be false during shutdown.
+			query.Execute(FALSE)
+			qdel(query)
+		for(var/datum/db_query/query in queries_active)
+			//Finish any remaining active qeries
+			UNTIL(query.process())
```

Comment:
https://github.com/tgstation/tgstation/pull/77311/files#diff-d0edfd88dc5eb1e319e6da1e72635d09dc5b35b930077ba34b67606bee32212cR600-R602 sadly there is one case where it will just always return without a true value and this will perma hang.

Reply:
also if this takes longer than 30 seconds for some stupid reason its probably better to just eat the crash.

Reply:
ah dick

Reply:
I've changed it to return True when the status actually is finished as well.

Reply:
i think ti's impossible for any queries to be on the queue with that state anyway, but better safe than sorry.